ouroboros-consensus-cardano-test/test/Test/Consensus/Cardano/Translation.hs

```diff
@@ -0,0 +1,380 @@
+import           Cardano.Ledger.Shelley.Translation
+import           Cardano.Ledger.Shelley.UTxO (UTxO (..))
+
+import           Ouroboros.Consensus.BlockchainTime.WallClock.Types
+                     (slotLengthFromSec)
+import           Ouroboros.Consensus.Cardano.Block (CardanoEras)
+import           Ouroboros.Consensus.Cardano.CanHardFork ()
+import           Ouroboros.Consensus.HardFork.Combinator (InPairs (..),
+                     hardForkEraTranslation, translateLedgerState)
+import           Ouroboros.Consensus.HardFork.Combinator.State.Types
+                     (TranslateLedgerState (translateLedgerStateWith))
+import           Ouroboros.Consensus.HardFork.Combinator.Util.InPairs
+                     (RequiringBoth, provideBoth)
+import           Ouroboros.Consensus.Ledger.Basics (LedgerCfg, LedgerConfig,
+                     LedgerState)
+import           Ouroboros.Consensus.Ledger.Tables
+import           Ouroboros.Consensus.Protocol.Praos
+import           Ouroboros.Consensus.Protocol.Praos.Common
+                     (MaxMajorProtVer (..))
+import           Ouroboros.Consensus.Protocol.Praos.Translate ()
+import           Ouroboros.Consensus.Protocol.TPraos (TPraos)
+import           Ouroboros.Consensus.TypeFamilyWrappers
+import           Ouroboros.Consensus.Util (dimap)
+
+import           Ouroboros.Consensus.Byron.Ledger (ByronBlock, byronLedgerState)
+
+import           Ouroboros.Consensus.Shelley.Eras
+import           Ouroboros.Consensus.Shelley.HFEras ()
+import           Ouroboros.Consensus.Shelley.Ledger (ShelleyBlock,
+                     ShelleyLedgerConfig, mkShelleyLedgerConfig,
+                     shelleyLedgerState, shelleyLedgerTables, shelleyUTxOTable)
+import           Ouroboros.Consensus.Shelley.Ledger.SupportsProtocol ()
+
+import           Test.Cardano.Ledger.Alonzo.Serialisation.Generators ()
+import           Test.Cardano.Ledger.Babbage.Serialisation.Generators ()
+import           Test.Cardano.Ledger.Conway.Serialisation.Generators ()
+import           Test.Cardano.Ledger.Shelley.Utils (unsafeBoundRational)
+import           Test.Consensus.Byron.Generators (genByronLedgerConfig,
+                     genByronLedgerState)
+import           Test.Consensus.Cardano.MockCrypto (MockCryptoCompatByron)
+import           Test.Consensus.Shelley.Generators ()
+import           Test.Consensus.Shelley.MockCrypto
+
+import           Test.Util.Time (dawnOfTime)
+
+-- Definitions to make the signatures a bit less unwieldy
+type Crypto = MockCryptoCompatByron
+type Proto  = TPraos Crypto
+
+tests :: TestTree
+tests = testGroup "UpdateTablesOnEraTransition"
+  [ testTablesTranslation "Byron to Shelley"
+                          byronToShelleyLedgerStateTranslation
+                          byronUtxosAreInsertsInShelleyUtxoDiff
+                          (\st -> cover 50 (      nonEmptyUtxosByron st) "UTxO set is not empty"
+                                -- FIXME: we should test with empyt UTxO!
```

Comment:
We could address this in this PR or as a follow-up issue.

Reply:
Done [#4473](https://github.com/input-output-hk/ouroboros-network/issues/4473)